blocks/default-output-block/output-types/default.jsx

```diff
@@ -84,11 +63,131 @@ const comscoreNoScript = (accountId) => {
+    })(window,document,'script','dataLayer','${gtmID}');
+  `;
+  return (
+    <>
+      <link rel="preconnect" href="https://www.googletagmanager.com/" />
+      <script data-integration="googleTagManager" dangerouslySetInnerHTML={{ __html: gtmScript }} />
+    </>
+  );
+};
+
+const googleTagManagerNoScript = (gtmID) => {
+  if (!gtmID) {
+    return null;
+  }
+  return (
+    <noscript>
+      <iframe
+        title="gtm"
+        src={`https://www.googletagmanager.com/ns.html?id=${gtmID}`}
+        height="0"
+        width="0"
+        style={{
+          display: 'none',
+          visibility: 'hidden',
+        }}
+      />
     </noscript>
   );
 };
 
+const nativoIntegrationScript = (nativoIntegration) => {
+  if (!nativoIntegration) {
+    return null;
+  }
+  return (
+    <script async data-integration="nativo-ad" src="https://s.ntv.io/serve/load.js" />
+  );
+};
+
+const querylyCode = (querylyId, querylyOrg, pageType) => {
+  if (!querylyId) {
+    return null;
+  }
+  const querylyInit = `
+    window.addEventListener('DOMContentLoaded', (event) => {
+      queryly.init("${querylyId}", document.querySelectorAll("#fusion-app"));
+    });
+  `;
+  return (
+    <>
+      <link rel="preconnect" href="https://www.queryly.com/" />
+      <script defer data-integration="queryly" src="https://www.queryly.com/js/queryly.v4.min.js" />
+      <script data-integration="queryly" dangerouslySetInnerHTML={{ __html: querylyInit }} />
+      { pageType === 'queryly-search'
+        ? <script defer data-integration="queryly" src={`https://www.queryly.com/js/${querylyOrg}-advanced-search.js`} />
+        : null}
+    </>
+  );
+};
+
+const fontUrlLink = (fontUrl) => {
+  // If fontURL is an array, then iterate over the array and build out the links
+  if (fontUrl && Array.isArray(fontUrl) && fontUrl.length > 0) {
+    const fontLinks = [...new Set(fontUrl)].map((url, index) => (
+      <>
+        <link rel="prefetch" as="font" key={url} data-testid={`font-loading-url-prefetch-${index}`} href={`${url}&display=swap`} crossOrigin="anonymous" />
+        <link rel="stylesheet" key={url} data-testid={`font-loading-url-${index}`} href={`${url}&display=swap`} />
+      </>
+    ));
+    return (
+      <>{fontLinks}</>
+    );
+  }
+  // Legacy support where fontUrl is a string
+  return fontUrl ? (
+    <>
+      <link rel="prefetch" as="font" href={`${fontUrl}&display=swap`} crossOrigin="anonymous" />
+      <link rel="stylesheet" href={`${fontUrl}&display=swap`} />
+    </>
+  ) : '';
+};
+
+const injectStringScriptArray = (scriptStringArray) => (
+  [...new Set(scriptStringArray)].map((scriptString, index) => (
+    // no good way of getting keys for this
+    // index used to remove warnings
+    // this key will not affect performance or issues with changing order
+    /* eslint-disable-next-line react/no-array-index-key */
+    <script key={index} data-integration="injected" dangerouslySetInnerHTML={{ __html: scriptString }} />
```

Comment:
Up to this we have 7 `dangerouslySetInnerHTML` and this one is also in a loop so we could have _n_ amount.

As they are all inline the parser has to stop and execute them as it comes across them - what if we collected all the scripts together and placed them in one single inline script?

Reply:
That's a good solution.  I just moved this logic from a different location, but I will adjust this.